### ISSUE TITLE
Error on line 23 - Logistic Regression

### ISSUE THREAD

**Issue by jacksusank** [rtealwitter/datamining2026spring#31]
I think you forgot to include the step function in the loss function:

Old Line: "Attempt #1: We could simply apply a step function to the output of the linear model, i.e., predict $y^{(i)} = 1$ if $\langle \mathbf{w}, \mathbf{x}^{(i)} \rangle > 0$ and $y^{(i)} = 0$ otherwise. The loss could be the difference between the predicted value and the true value, i.e., $\mathcal{L}(\mathbf{w}) = \sum_{i=1}^n |y^{(i)} - \langle \mathbf{w}, \mathbf{x}^{(i)} \rangle|$. However, this loss is not differentiable, so we cannot use gradient descent to optimize it."

Proposed Fix: "Attempt #1: We could simply apply a step function to the output of the linear model, i.e., predict $y^{(i)} = 1$ if $\langle \mathbf{w}, \mathbf{x}^{(i)} \rangle > 0$ and $y^{(i)} = 0$ otherwise. The loss could be the difference between the predicted value and the true value, i.e., $\mathcal{L}(\mathbf{w}) =  \sum_{i=1}^n \left| y^{(i)} - \mathbf{1}[\langle \mathbf{w}, \mathbf{x}^{(i)} \rangle > 0] \right|$ . However, this loss is not differentiable, so we cannot use gradient descent to optimize it."

https://github.com/rtealwitter/datamining2026spring/blob/4ab71393f59e05fd1ad92b1a427756007888f03a/notes/LogisticRegression.qmd#L23

**Comment by rtealwitter** [comment]
This was intentional
